sunbeam-python/sunbeam/plugins/validation/plugin.py

```diff
@@ -0,0 +1,469 @@
+# Copyright (c) 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import subprocess
+from datetime import datetime
+from pathlib import Path
+from typing import Any, Dict, List, Optional
+
+import click
+import pydantic
+from croniter import croniter
+from packaging.version import Version
+from rich.console import Console
+
+from sunbeam.clusterd.service import ClusterServiceUnavailableException
+from sunbeam.commands.openstack import OPENSTACK_MODEL
+from sunbeam.jobs.deployment import Deployment
+from sunbeam.jobs.juju import JujuHelper, run_sync
+from sunbeam.jobs.plugin import PluginManager
+from sunbeam.plugins.interface.v1.openstack import (
+    OpenStackControlPlanePlugin,
+    TerraformPlanLocation,
+)
+
+LOG = logging.getLogger(__name__)
+console = Console()
+
+PLUGIN_VERSION = "0.0.1"
+MINIMAL_PERIOD = 15 * 60  # 15 minutes in seconds
+TEMPEST_APP_NAME = "tempest"
+TEMPEST_CONTAINER_NAME = "tempest"
+TEMPEST_CHANNEL = "2023.2/edge"
```

Comment:
Can you also move TEMPEST_CHANNEL to sunbeam-python/sunbeam/versions.py
Single place to change channels for core and in-built plugins.

Reply:
Done